modules/IntegrationTesting/src/scenarioexecution.cpp

```diff
@@ -0,0 +1,154 @@
+
+/**
+ * @brief An integration test for scenario execution. This test will run a scenario execution for one object,
+ * and check that all states are correct and that it follows the trajectory.
+ * 
+ */
+ScenarioExecution::ScenarioExecution() : IntegrationTesting(testName) {
+	RCLCPP_INFO(get_logger(), "Started integration test: %s", testName.c_str());
+	getObjectTrajectoryClient = this->create_client<atos_interfaces::srv::GetObjectTrajectory>("/atos/get_object_trajectory");
+	runIntegrationTest();
+}
+
+
+/**
+ * @brief Destructor.
+ * 
+ */
+ScenarioExecution::~ScenarioExecution() {}
+
+
+/**
+ * @brief Run the integration test. This integration test will initialize a scenario, connect to object 1,
+ * arm the object, and then start it. It will check that all states are correct, and that the object follows
+ * the trajectory.
+ * 
+ */
+void ScenarioExecution::runIntegrationTest() {
+	RCLCPP_INFO(get_logger(), "Running scenario");
+	auto msg = std_msgs::msg::Empty();
+
+	initPub->publish(msg);
+	std::this_thread::sleep_for(1s); // sleep to allow the object to initialize
+	checkState(initTopic);
+	
+	connectPub->publish(msg);
+	std::this_thread::sleep_for(1s); // sleep to allow the object to connect
+	checkState(connectTopic);
+
+	armPub->publish(msg);
+	std::this_thread::sleep_for(1s); // sleep to allow the object to arm
+	checkState(armTopic);
+
+	startPub->publish(msg);
+	std::this_thread::sleep_for(1s); // sleep to allow the object to start
+	checkState(startTopic);
+
+	checkTrajectory();
+	printResult();
+}	
+
+
+/**
+ * @brief Get the x- and y-coordinates of the trajectory points for object 1.
+ * 
+ * @return std::vector<std::pair<double, double>> Trajectory points.
+ */
+std::vector<std::pair<double, double>> ScenarioExecution::getTrajectoryPoints() {
+	auto request = std::make_shared<atos_interfaces::srv::GetObjectTrajectory::Request>();
+	request->id = 1;
+	std::shared_ptr<atos_interfaces::srv::GetObjectTrajectory::Response> response;
+	this->callService(1000ms, getObjectTrajectoryClient, response, request);
+
+	std::vector<std::pair<double, double>> trajectory;
+	for (const auto& t : response->trajectory.points) {
+		trajectory.push_back(std::make_pair(t.pose.position.x, t.pose.position.y));
+	}
+	return trajectory;
+}
+
+
+/**
+ * @brief Check that the object ends up near the end point of the trajectory.
+ * 
+ */
+void ScenarioExecution::checkTrajectory() {
+	auto trajectory = getTrajectoryPoints();
+	std::pair<double, double> lastPoint;
+	auto isObjectMoving = true;
+	while(isObjectMoving) {
+		atos_interfaces::msg::Monitor monr;
+		monitorSub = this->create_subscription<atos_interfaces::msg::Monitor>("/atos/object_1/object_monitor", 10, std::bind(&ScenarioExecution::placeholderCallback, this, std::placeholders::_1));
+		rclcpp::wait_for_message(monr, monitorSub, this->get_node_base_interface()->get_context(), 50ms);
+		if (monr.velocity.twist.linear.x == 0 && monr.velocity.twist.linear.y == 0) {
```

Comment:
You should probably add some leeway to the check here since objects might not always report exactly 0 velocity.
```suggestion
                speedThreshold = 0.1
		if (abs(monr.velocity.twist.linear.x) < speedThreshold && abs(monr.velocity.twist.linear.y) < speedThreshold) {
```